graph/connected_components_with_dsu.cpp

```diff
@@ -1,56 +1,112 @@
-#include <iostream>
-#include <set>
-#include <vector>
+/**
+ * @file
+ * @brief [Disjoint union](https://en.wikipedia.org/wiki/Disjoint_union)
+ *
+ * @details
+ * The Disjoint union is the technique to find connected component in graph efficiently.
+ *
+ * ### Algorithm
+ * In Graph, if you have to find out the number of connected components, there are 2 options
+ * 1. Depth first search
+ * 2. Disjoint union
+ * 1st option is inefficient, Disjoint union is the most optimal way to find this.
+ */
+#include <iostream> /// for io operations
+#include <set>    /// for std::set
+#include <vector> /// for std::vector
 
-int number_of_nodes;  // denotes number of nodes;
-std::vector<int> parent;
-std::vector<int> connected_set_size;
-void make_set() {  // function the initialize every node as it's own parent
-    for (int i = 1; i <= number_of_nodes; i++) {
+/**
+ * @namespace graph
+ * @brief Graph Algorithms
+ */
+namespace graph {
+
+/**
+ * @namespace disjoint_union
+ * @brief Function for [Disjoint union] (https://en.wikipedia.org/wiki/Disjoint_union) implementation
+ */
+namespace disjoint_union {
+
+int64_t number_of_nodes;                  // denotes number of nodes
+std::vector<int64_t> parent;              // parent of each node
+std::vector<int64_t> connected_set_size;  // size of each set
+/**
+ * @brief function the initialize every node as it's own parent
+ * @returns void
+ */
+void make_set() {
+    for (int64_t i = 1; i <= number_of_nodes; i++) {
         parent[i] = i;
         connected_set_size[i] = 1;
     }
 }
-// To find the component where following node belongs to
-int find_set(int v) {
-    if (v == parent[v]) {
-        return v;
+/**
+ * @brief To find the component where following node belongs to
+ * @param val parent of val should be found
+ * @return parent of val
+ */
+int64_t find_set(int64_t val) {
+    while (parent[val] != val) {
+        parent[val] = parent[parent[val]];
+        val = parent[val];
     }
-    return parent[v] = find_set(parent[v]);
+    return val;
 }
+/**
+ * @brief To join 2 components to belong to one
```

Comment:
```suggestion
 * @brief Merge two components to become one
```

Reply:
I am not a native speaker of English, that is why this kind of mistake happens.